common/scripts/prep-samples.js

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+const fs = require("fs");
+const path = require("path");
+
+// Accept a base directory (package directory) as an argument or use CWD
+const args = process.argv.slice(2);
+
+let baseDir;
+if (args.length > 0) {
+  baseDir = path.resolve(args[0]);
+} else {
+  baseDir = process.cwd();
+}
+const tsDir = path.join(baseDir, "samples", "typescript");
+
+/**
+ * Breadth-first search for files ending in .ts
+ */
+function findAllTsFiles(filesAndFolders) {
+  const q = filesAndFolders.map(function(f) {
+    return [f, tsDir];
+  });
+  const tsFiles = [];
+
+  let idx = 0;
+
+  while (idx !== q.length) {
```

Comment:
this feels a little odd to me. How about `while (q.length)` and use `q.shift()` to take from the front of the array?